cmd/simulator/load/worker.go

```diff
@@ -44,16 +51,34 @@ func NewSingleAddressTxWorker(ctx context.Context, client ethclient.Client, addr
 	return tw
 }
 
-func (tw *singleAddressTxWorker) IssueTx(ctx context.Context, tx *types.Transaction) error {
-	return tw.client.SendTransaction(ctx, tx)
+func (tw *singleAddressTxWorker) IssueTx(ctx context.Context, timeTx txs.TimeTx) error {
+	start := time.Now()
+	err := tw.client.SendTransaction(ctx, timeTx.Tx)
+	issuanceDuration := time.Since(start)
+	if err != nil {
+		return err
+	}
+
+	timeTx.IssuanceStart = start
+	timeTx.IssuanceDuration = issuanceDuration
+	tw.issuanceHistogram = append(tw.issuanceHistogram, timeTx.IssuanceDuration.Seconds())
+	return nil
 }
 
-func (tw *singleAddressTxWorker) ConfirmTx(ctx context.Context, tx *types.Transaction) error {
+func (tw *singleAddressTxWorker) ConfirmTx(ctx context.Context, timeTx txs.TimeTx) error {
+	tx := timeTx.Tx
 	txNonce := tx.Nonce()
 
+	start := time.Now()
 	for {
 		// If the is less than what has already been accepted, the transaction is confirmed
 		if txNonce < tw.acceptedNonce {
+			confirmationEnd := time.Now()
+			timeTx.ConfirmationDuration = confirmationEnd.Sub(start)
+			timeTx.IssuanceToConfirmationDuration = confirmationEnd.Sub(timeTx.IssuanceStart)
+			tw.issuanceToConfirmationHistogram = append(tw.issuanceToConfirmationHistogram, timeTx.IssuanceToConfirmationDuration.Seconds())
+			tw.confirmationHistogram = append(tw.confirmationHistogram, timeTx.ConfirmationDuration.Seconds())
```

Comment:
since we confirm txs in batches sequentially, if one tx is taking long to confirm, this will skew the metrics.
not sure if this is okay for our intended use case here.
We could alternatively spawn threads to wait on txs individually or subscribe to a feed.
If we switch to non-sequential confirm time measurement, this part will need synchronization.

Reply:
I think we want to show all individual tx times. We want to see the ones that took a while too. Maybe we should show this in terms of percentile instead of a histogram.  cc: @aaronbuchwald